content/post/announcing-pga.md

```diff
@@ -0,0 +1,129 @@
+---
+author: campoy
+date: 2018-06-05
+title: "Announcing Public Git Archive"
+image: /post/announcing-pga/pga.png
+description: "Announcing Public Git Archive, the largest git repositories dataset in the world."
+categories: ["technical", "MLonCode"]
+---
+
+Last week we had the honor of participating at [MSR'18](https://2018.msrconf.org/),
+where two of the members of our team, [Vadim Markovtsev](https://twitter.com/vadimlearning)
+and [Waren Long](https://twitter.com/warenlg), presented the
+[research paper](https://arxiv.org/abs/1803.10144) they wrote on our latest dataset:
+`Public Git Archive`.
+
+`Public Git Archive` is the result of months of effort curating a dataset suitable for
+training Machine Learning on Source Code (aka MLonCode) models.
+
+## Dataset Contents
+
+The dataset contains 3TB of repositories from GitHub ready to download.
+This includes all of the contents (git metadata and file contents) for all of the repositories
+on GitHub with 50 or more stars.
+The list of repositories was obtained from [the GHTorrent project](http://ghtorrent.org/),
+which provides information about GitHub repositories but unfortunately doesn't include the file contents.
+
+### Rooted Repositories
+
+To avoid unnecessary repeated files we unified all of the repositories by merging all of the
+git trees which contained a common root, according to their commit hashes.
+There is still file repetition when two or more repositories contain copies of a single file,
+but have no common root.
+
+The figure below shows visually two repositories being merged into a single one, or rooted.
+
+![rooted repositories](/post/announcing-pga/rooted-repos.png)
+
+Just to give you an idea of the size of the dataset, let's compare it to previous
+datasets:
+
+ _  |  Qualitas Corpus  |  Sourcerer  |  GitHub Java Corpus  |  Public Git Archive  |
+---|-------------------|-------------|----------------------|----------------------|
+N projects | 111 | 19,233 | 14,807 | 182,014 |
+Year of release | 2013 | 2014 | 2013 | 2018 |
+Languages | 1 (Java) | 1 (Java) | 1 (Java) | 455 |
+Repeatable | No | No | No | Yes |
+N files | 177k | 1.9M | 1.5M | 54.5M (HEAD) |
+Lines of Code | 37M | 320M | 352M | 15,941M (HEAD) |
+Storage Size | 1.3GB | 19GB | 14GB | 3.0TB |
+
+And since an image is worth a thousand tables, here you can see the size of Public Git Archive
+compared to other datasets. _Note that the scale is logarithmic!_
+
+![size of datasets](/post/announcing-pga/size.png)
+
+Similarly, the number of lines of code show the size of Public Git Archive.
+
+![lines of code of datasets](/post/announcing-pga/loc.png)
```

Comment:
We havae had a popular question: compare to to BigQuery.

The answer is: BigQuery contains only the HEAD revision, there is no way to get an arbitrary revision, thus is excluded from the comparison. The second difference is that the Git commit metadata is not anonymized.

Then people will ask WTH privacy leak WAT FUUUU. The answer is, we are not violating the US laws and the GDPR contains a handy exclusion: https://news.ycombinator.com/item?id=16509688

Reply:
I'm not sure where this should go, tbh. I can add some mention in the licensing section, or simply leave it out.

Reply:
We had this question on MSR as you remember. We will have to answer it, either in Reddit comments or here.

Reply:
Oh, I missed this answer. I'll send a PR with a clarification about BigQuery.
Maybe we should add a FAQ in pga.sourced.tech (which definitely needs improvement)

Reply:
Agree with the BQ part needed